wicker/core/storage.py

```diff
@@ -27,7 +29,62 @@
 logger = logging.getLogger(__name__)
 
 
-class S3DataStorage:
+class AbstractDataStorage(ABC):
+    """Abstract data storage class that implements required methods for Column Bytes File Cache"""
+
+    @abstractmethod
+    def fetch_file(self, input_path: str, local_prefix: str, timeout_seconds: int) -> str:
+        """Fetch file from chosen data storage method.
+
+        :param input_path: input file path
+        :param local_prefix: local path that specifies where to download the file
+        :param timeout_seconds: number of seconds till timing out on waiting for the file to be downloaded
+        :return: local path to the downloaded file
+
+        """
+        pass
+
+
+class LocalDataStorage(AbstractDataStorage):
+    """Storage routines for reading and writing objects in local fs"""
+
+    def __init__(self) -> None:
+        """Constructor for a local FS"""
+        pass
+
+    @retry(
+        Exception,
+        tries=get_config().storage_download_config.retries,
+        backoff=get_config().storage_download_config.retry_backoff,
+        delay=get_config().storage_download_config.retry_delay_s,
+        logger=logger,
+    )
+    def download_with_retries(self, input_path: str, local_path: str):
+        try:
+            shutil.copyfile(input_path, local_path)
```

Comment:
hmm, Why do we need to copy in local file system?

Reply:
@aalavian it's true, for FSx-style systems (including EFS) the remote filesystem appears as a local volume and for users it is not obvious that they are paying a network latency penalty for every access to these files.

In another one of our internal ML frameworks in Japan (that begins "In"), the data access pattern is actually to leave the files in FSx for every epoch of training, so that the network penalty is always paid. For most of the models, this might actually be fine as the data is often accessible by the time the GPU is ready. However, it would be nice to have the option to copy the data from the locally-mounted (remote) volume to a fast local SSD.

For the FUSE filesystems for bucket storage from GCP and AWS, they actually have internal caching options so that this behavior is effectively done by these FUSE drivers, so in that case we will not call this function.

Reply:
Thanks for the clarification folks, I think `LocalDataStorage` and in particular doc-strings in this class referring to `local file system` would raise the same question for others.

How about we renaming this class and doc string to be more generic than local, maybe something like `FileSystemDataStorage` and create an Github issue to add the option to copy or not so we would not forget it?